packages/user/src/services/user-module.ts

```diff
@@ -41,29 +46,37 @@ export default class UserModuleService<TUser extends User = User>
   protected baseRepository_: DAL.RepositoryService
 
   protected readonly userService_: ModulesSdkTypes.InternalModuleService<TUser>
+  protected readonly inviteService_: ModulesSdkTypes.InternalModuleService<TInvite>
 
   constructor(
-    { userService, baseRepository }: InjectedDependencies,
+    { userService, inviteService, baseRepository }: InjectedDependencies,
     protected readonly moduleDeclaration: InternalModuleDeclaration
   ) {
     // @ts-ignore
     super(...arguments)
 
     this.baseRepository_ = baseRepository
     this.userService_ = userService
+    this.inviteService_ = inviteService
   }
 
-  create(data: CreateUserDTO[], sharedContext?: Context): Promise<UserDTO[]>
-  create(data: CreateUserDTO, sharedContext?: Context): Promise<UserDTO>
+  create(
+    data: UserTypes.CreateUserDTO[],
+    sharedContext?: Context
+  ): Promise<UserTypes.UserDTO[]>
+  create(
+    data: UserTypes.CreateUserDTO,
+    sharedContext?: Context
+  ): Promise<UserTypes.UserDTO>
 
-  @InjectManager("baseRepository_")
+  @InjectTransactionManager("baseRepository_")
   async create(
-    data: CreateUserDTO[] | CreateUserDTO,
+    data: UserTypes.CreateUserDTO[] | UserTypes.CreateUserDTO,
     @MedusaContext() sharedContext: Context = {}
   ): Promise<UserTypes.UserDTO | UserTypes.UserDTO[]> {
     const input = Array.isArray(data) ? data : [data]
 
-    const users = await this.create_(input, sharedContext)
+    const users = await this.userService_.create(input, sharedContext)
```

Comment:
comment: Just to ensure everybody is aware, this does not work when relations are involved. In that case it is better to have the create calling the create_ and then serialize in the create. Otherwise the populate could not function properly. wdyt?

cc @olivermrbl

Reply:
same for update

Reply:
I am not sure I understand. As long as we are not doing anything but creating/updating, shouldn't this be fine?